lib/libc/newlib/CMakeLists.txt

```diff
@@ -27,3 +27,10 @@ zephyr_link_libraries(
   $<$<BOOL:${CONFIG_NEWLIB_LIBC_FLOAT_SCANF}>:-u_scanf_float>
   gcc # Lib C depends on libgcc. e.g. libc.a(lib_a-fvwrite.o) references __aeabi_idiv
   )
+
+if(CONFIG_NEWLIB_LIBC_NANO)
+  zephyr_link_libraries(
+    -specs=nano.specs
+    )
+endif()
+
```

Comment:
You probably want to pass that flag also for compiling using `zephyr_compile_options`, so that the nano headers defined in the spec file are used instead of the default ones.

Note however that it is currently broken by PR #13538

Reply:
It seems to me that " --specs=nano.specs" is only a linker option.
Please see [Nano libraries usage](http://www.openstm32.org/Other%2BManuals#Nano_libraries_usage)

Reply:
I haven't looked at the documentation, but at the content of `nano.specs` on my system

```
*cpp_unique_options:
-isystem /usr/include/newlib/nano %(nano_cpp_unique_options)
```

This adds a new system include path pointing to `/usr/include/newlib/nano`. This directory contains a single file `newlib.h` containing the newlib configuration. In the nano version a few options have been disabled. Some of those defines are used by other newlib headers changing the signature of some functions, so I believe this is something important to do.

Note however that the Zephyr SDK 0.10.0 provides a `nano.specs` file, but the include path added by this file is not present. It looks like a packaging issue, I'll investigate and fill a bug.

Reply:
See also https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=801338

Reply:
I added the "zephyr_compile_options" into the "CMakeLists.txt"